lib/terrafying/components/letsencrypt.rb

```diff
@@ -355,17 +355,47 @@ def renew
 
         resource :aws_cloudwatch_event_target, "#{@name}_lambda_event_target", {
           rule: event_rule["name"],
-          target_id: lamda_function["id"],
-          arn: lamda_function["arn"]
+          target_id: lambda_function["id"],
+          arn: lambda_function["arn"]
         }
 
         resource :aws_lambda_permission, "allow_cloudwatch_to_invoke_#{@name}_lambda", {
           statement_id: "AllowExecutionFromCloudWatch",
           action: "lambda:InvokeFunction",
-          function_name: lamda_function["function_name"],
+          function_name: lambda_function["function_name"],
           principal: "events.amazonaws.com",
           source_arn: event_rule["arn"]
         }
+
+        resource :aws_sns_topic, "#{@name}_lambda_cloudwatch_topic",
+                 name: "#{@name}_lambda_cloudwatch_topic"
+
+        resource :aws_cloudwatch_metric_alarm, "#{@name}_lambda_failure_alarm",
+                 alarm_name: "#{@name}-lambda-failure-alarm",
+                 comparison_operator: "GreaterThanOrEqualToThreshold",
+                 evaluation_periods: "1",
+                 period: "300",
+                 metric_name: "Errors",
+                 namespace: "AWS/Lambda",
+                 threshold: 1,
+                 alarm_description: "Alert generated if the #{@name} certbot lambda fails execution",
+                 actions_enabled: true,
+                 dimensions: {
+                           FunctionName: lambda_function["function_name"]
+                         },
+                 alarm_actions: [
+                           "${aws_sns_topic.#{@name}_lambda_cloudwatch_topic.arn}"
+                         ],
+                 ok_actions: [
+                          "${aws_sns_topic.#{@name}_lambda_cloudwatch_topic.arn}"
+                        ]
+
+        resource :aws_sns_topic_subscription, "#{@name}_lambda_cloudwatch_subscription",
+                 topic_arn: "${aws_sns_topic.#{@name}_lambda_cloudwatch_topic.arn}",
+                 protocol: "https",
+                 endpoint: "https://events.pagerduty.com/integration/7cd0edaa372843dea1ee7bfb4b930c6a/enqueue",
```

Comment:
is this generic? or very specific to us? considering this an open source repo?

Reply:
ah I forgot - sorry, that is a very-specific to us pd integration which is now deleted

Reply:
Will find another way to get a PD integration key in there